### PR TITLE
xeval/evaluator: add math function

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR add math function `abs_int` to expression evaluator.

PTAL @XuHuaiyu @AndreMouche @lishihai9017 @andelf

### CODE REVIEW

File: src/util/xeval/evaluator.rs [hunk 1]
```diff
@@ -223,6 +224,15 @@ impl Evaluator {
         left.cmp(ctx, &right).map(Some).map_err(From::from)
     }
 
+    pub fn get_one_child<'a>(&mut self, expr: &'a Expr) -> Result<&'a Expr> {
+        let l = expr.get_children().len();
+        if l != 1 {
+            return Err(Error::Expr(format!("{:?} need 1 operand but got {}", expr.get_tp(), 1)));
```
Comment (andelf) [inline src/util/xeval/evaluator.rs]: last `1` is wrong

**Comment by hhkbp2** [comment]
PTAL @XuHuaiyu @AndreMouche @lishihai9017 @andelf

**Comment by hhkbp2** [comment]
The impl of `abs_int` will be raised as a new PR, so that it would be a independent  PR for the demonstration in document.

File: src/util/xeval/evaluator.rs [hunk 2]
```diff
@@ -513,6 +518,13 @@ impl Evaluator {
         Ok(Datum::Json(first.merge(suffixes)))
     }
 
+    fn eval_scalar_function(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        match expr.get_sig() {
+            ScalarFuncSig::AbsInt => self.abs_int(ctx, expr),
```
Comment (lishihai9017) [inline src/util/xeval/evaluator.rs]: removing branch? since impl had removed.

**Comment by lishihai9017** [comment]
LGTM

File: src/util/xeval/evaluator.rs (hunk 2 above)
Comment (hhkbp2) [inline src/util/xeval/evaluator.rs]: I think we should add all these "framework" stuff, such as protocol to function call mapping, function definition stubs, e.g. before letting the community join in. It's easier for anyone to just implement the stub function(which is `abs_int` here) without add other code.

**Pull request merged**